components/kyma-environment-broker/internal/process/deprovisioning/event_hub_step.go

```diff
@@ -0,0 +1,102 @@
+package deprovisioning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/hyperscaler/azure"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-incubator/compass/components/kyma-environment-broker/internal/storage"
+)
+
+type DeprovisionAzureEventHubStep struct {
+	operationManager *process.DeprovisionOperationManager
+	hyperscalerProvider azure.HyperscalerProvider
+	instanceStorage  storage.Instances
+	accountProvider  hyperscaler.AccountProvider
+	context context.Context
+}
+
+func NewDeprovisionAzureEventHubStep(os storage.Operations, is storage.Instances,hyperscalerProvider azure.HyperscalerProvider, accountProvider hyperscaler.AccountProvider, ctx context.Context) DeprovisionAzureEventHubStep {
+	return DeprovisionAzureEventHubStep{
+		operationManager:  process.NewDeprovisionOperationManager(os),
+		hyperscalerProvider: hyperscalerProvider,
+		instanceStorage:     is,
+		accountProvider:     accountProvider,
+		context:             ctx,
+	}
+}
+
+var _ Step = (*DeprovisionAzureEventHubStep)(nil)
+
+func (s DeprovisionAzureEventHubStep) Name() string {
+	return "Deprovision Azure Event Hubs"
+}
+
+func (s DeprovisionAzureEventHubStep) Run(operation internal.DeprovisioningOperation, log logrus.FieldLogger) (internal.DeprovisioningOperation, time.Duration, error) {
+	// TODO(nachtmaar): global timeout as in remove_runtime.go ?
+	// TODO(nachtmaar): have shared code between provisioning/deprovisioning
+	hypType := hyperscaler.Azure
+
+	// parse provisioning parameters
+	pp, err := operation.GetParameters()
+	if err != nil {
+		// if the parameters are incorrect, there is no reason to retry the operation
+		// a new request has to be issued by the user
+		log.Errorf("Aborting after failing to get valid operation provisioning parameters: %v", err)
+		return s.operationManager.OperationFailed(operation, "invalid operation provisioning parameters")
+	}
+	log.Infof("HAP lookup for credentials to provision cluster for global account ID %s on Hyperscaler %s", pp.ErsContext.GlobalAccountID, hypType)
+
+
+	//instance, err := s.instanceStorage.GetByID(operation.InstanceID)
```

Comment:
@nachtmaar  No need so far for the instance as long as we have the provisioning parameters to get the credentials

Reply:
We brought it back :-D